scripts/zones/Misareaux_Coast/npcs/qm2.lua

```diff
@@ -0,0 +1,62 @@
+-----------------------------------
+-- Area: Misareaux_Coast
+-- NPC:  ??? (Spawn Ziphius)
+-- !pos 76 -16 534 25
+-----------------------------------
+package.loaded["scripts/zones/Misareaux_Coast/TextIDs"] = nil;
+-----------------------------------
+require("scripts/zones/Misareaux_Coast/TextIDs");
+require("scripts/zones/Misareaux_Coast/MobIDs");
+require("scripts/globals/npc_util");
+
+function onTrade(player,npc,trade)
+    local baited = GetServerVariable("[Ziphius]Bait Trap");
+
+    -- Trade Slice of Carp
+    if (npcUtil.tradeHas(trade, 16994) and bit.band(baited, bit.lshift(1, (npc:getID() - ZIPHIUS_QM_BASE))) == 0) then
+        baited = bit.bor(baited, bit.lshift(1, (npc:getID() - ZIPHIUS_QM_BASE)));
+        player:tradeComplete();
```

Comment:
if using npcUtil.tradeHas() to check for traded items, the transaction should be completed with player:confirmTrade() rather than player:tradeComplete().  This way, only the required items will be removed from the player, and any excess ones will be returned.  (pending merge of #4431 )